index.js

```diff
@@ -11,8 +9,50 @@ const randomstring = require('randomstring');
 const requestretry = require('requestretry');
 const tinytim = require('tinytim');
 const yaml = require('js-yaml');
+const _ = require('lodash');
+
+const MAXATTEMPTS = 5;
+const RETRYDELAY = 3000;
 const CPU_RESOURCE = 'beta.screwdriver.cd/cpu';
 const RAM_RESOURCE = 'beta.screwdriver.cd/ram';
+const TOLERATIONS_PATH = 'spec.tolerations';
+const AFFINITY_NODE_SELECTOR_PATH = 'spec.affinity.nodeAffinity.' +
+    'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms';
+
+/**
+ * Parses nodeSelector config and update intended nodeSelector in tolerations
+ * and nodeAffinity.
+ * @method setNodeSelector
+ * @param {Object} podConfig k8s pod config
+ * @param {Object} nodeSelectors key-value pairs of node selectors
+ */
+function setNodeSelector(podConfig, nodeSelectors) {
+    if (!nodeSelectors || typeof nodeSelectors !== 'object') {
+        return;
+    }
+
+    const tolerations = _.get(podConfig, TOLERATIONS_PATH, []);
+    const nodeAffinitySelectors = _.get(podConfig, AFFINITY_NODE_SELECTOR_PATH, []);
+
+    Object.keys(nodeSelectors).forEach((key) => {
+        tolerations.push({
+            key,
```

Comment:
should this be `'key': key`
an example yaml: 
```apiVersion: v1
kind: Pod
metadata:
  name: testpod-dedicated
  namespace: yby-chestery-k8s
  labels:
    env: test
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: dedicated
            operator: In
            values:
            - screwdriver
  tolerations:
  - key: dedicated
    operator: Equal
    value: screwdriver
    effect: NoSchedule
  containers:
    - name: rhel6-base
      image: docker-```

Reply:
yes, `{ key }` will expand to `{'key', key}`

Reply:
arh, interesting.